tests/test_metadata_eq_.py

```diff
@@ -41,34 +37,74 @@ def setUpClass(cls) -> None:
         cls.repo_dir = os.path.join(
             utils.TESTS_DIR, "repository_data", "repository", "metadata"
         )
+        cls.objects = {}
         cls.metadata = {}
         for md in TOP_LEVEL_ROLE_NAMES:
             with open(os.path.join(cls.repo_dir, f"{md}.json"), "rb") as f:
-                cls.metadata[md] = f.read()
+                data = f.read()
+                cls.metadata[md] = data
+                md_name = md[0].upper() + md[1:]
+                cls.objects[md_name] = Metadata.from_bytes(data).signed
+
+        cls.objects["Metadata"] = Metadata(cls.objects["Timestamp"], {})
+        cls.objects["Signed"] = cls.objects["Timestamp"]
+        cls.objects["Key"] = Key(
+            "id", "rsa", "rsassa-pss-sha256", {"public": "foo"}
+        )
+        cls.objects["Role"] = Role(["keyid1", "keyid2"], 3)
+        cls.objects["MetaFile"] = MetaFile(1, 12, {"sha256": "abc"})
+        cls.objects["DelegatedRole"] = DelegatedRole("a", [], 1, False, ["d"])
+        cls.objects["Delegations"] = Delegations(
+            {"keyid": cls.objects["Key"]}, {"a": cls.objects["DelegatedRole"]}
+        )
+        cls.objects["TargetFile"] = TargetFile(
+            1, {"sha256": "abc"}, "file1.txt"
+        )
+
+    # Keys are class names.
+    # Values are a dictionary. In those dictionaries their keys are class
+    # attributes and values are the new attribute worth.
+    classes_attributes: utils.DataSet = {
+        "Metadata": {"signed": None, "signatures": None},
+        "Signed": {"version": -1, "spec_version": "0.0.0"},
+        "Key": {"keyid": "a", "keytype": "foo", "scheme": "b", "keyval": "b"},
+        "Role": {"keyids": [], "threshold": 10},
+        "Root": {"consistent_snapshot": None, "keys": {}},
+        "MetaFile": {"version": None, "length": None, "hashes": {}},
+        "Timestamp": {"snapshot_meta": None},
+        "Snapshot": {"meta": None},
+        "DelegatedRole": {
+            "name": "",
+            "terminating": None,
+            "paths": [""],
+            "path_hash_prefixes": [""],
+        },
+        "Delegations": {"keys": {}, "roles": {}},
+        "TargetFile": {"length": 0, "hashes": {}, "path": ""},
+        "Targets": {"targets": {}, "delegations": []},
+    }
+
+    @utils.run_sub_tests_with_dataset(classes_attributes)
+    def test_classes_eq_(self, test_case_data: Dict[str, Any]) -> None:
+        obj = self.objects[self.case_name]  # pylint: disable=no-member
 
-    def copy_and_simple_assert(self, obj: Any) -> Any:
         # Assert that obj is not equal to an object from another type
         self.assertNotEqual(obj, "")
-        result_obj = copy.deepcopy(obj)
+        obj_2 = copy.deepcopy(obj)
         # Assert that __eq__ works for equal objects.
-        self.assertEqual(obj, result_obj)
-        return result_obj
+        self.assertEqual(obj, obj_2)
 
```

Comment:
Is it worth adding a oneliner comment about what you are doing here? This part of the test feels like the most magical, yet you only commented the simpler ones above. :)